api/contents.go

```diff
@@ -14,6 +14,14 @@ import (
 	"gorm.io/gorm"
 )
 
+type ContentCollection struct {
+	CommP      string `json:"commp"`
+	PayloadCID string `json:"payload_cid"`
+	Size       uint64 `json:"size"`
+	PaddedSize uint64 `json:"padded_size"`
+	Collection string `json:"collection"`
```

Comment:
It also needs to take the `ContentLocation` i.e, `content_location`, which just gets passed through to the new db record - see it in `db/schema.go`

This field is technically optional, but just make it mandatory for now since the only use case requires it

Reply:
<Could you also update the existing /contents/:dataset endpoint to take the ContentLocation too - make it optional there however>

Reply:
added content_location to required field for `/contents`. Its already a optional field for `/contents/:dataset` under the `db.Content` struct.